File: app/src/main/java/com/example/myapplication/ui/home/HomeFragment.java
package com.example.myapplication.ui.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import com.example.myapplication.R;
import com.example.myapplication.databinding.FragmentHomeBinding;

public class HomeFragment extends Fragment {

    private FragmentHomeBinding binding;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {

        binding = FragmentHomeBinding.inflate(inflater, container, false);
        View root = binding.getRoot();

        ImageButton button1 = root.findViewById(R.id.button1);
        ImageButton button2 = root.findViewById(R.id.button2);
        ImageButton button3 = root.findViewById(R.id.button3);

        button1.setOnClickListener(view -> {
            // Handle button1 click event
            // Display products with type "bags" using a gallery
            Bundle bundle = new Bundle();
            bundle.putString("productType", "bags");
            // Navigation.findNavController(view).navigate(R.id.action_nav_home_to_nav_gallery, bundle);
        });

        button2.setOnClickListener(view -> {
            // Handle button2 click event
            // Display products with type "gloves" using a gallery
            Bundle bundle = new Bundle();
            bundle.putString("productType", "gloves");
            // Navigation.findNavController(view).navigate(R.id.action_nav_home_to_nav_gallery, bundle);
        });

        button3.setOnClickListener(view -> {
            // Handle button3 click event
            // Display products with type "wallets" using a gallery
            Bundle bundle = new Bundle();
            bundle.putString("productType", "wallets");
            // Navigation.findNavController(view).navigate(R.id.action_nav_home_to_nav_gallery, bundle);
        });

        return root;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/example/product_tracker/ui/home/HomeFragment.kt
package com.example.product_tracker.ui.home

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import androidx.fragment.app.Fragment
import com.example.product_tracker.R
import com.example.product_tracker.databinding.FragmentHomeBinding
import com.example.product_tracker.model.ProductTypeMapper

class HomeFragment : Fragment() {
    private var binding: FragmentHomeBinding? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View {
        binding = FragmentHomeBinding.inflate(inflater, container, false)
        val root: View = binding!!.root

        val typeValues = ProductTypeMapper(requireContext()).getAllTypeValues()

        val buttons = listOf(
            Pair(R.id.button_bags, typeValues[0]),
            Pair(R.id.button_wallets, typeValues[1]),
            Pair(R.id.button_gloves, typeValues[2]),
            Pair(R.id.button_suitcases,typeValues[3]),
            Pair(R.id.button_belts, typeValues[4]),
            Pair(R.id.button_accessories, typeValues[5])
        )

        buttons.forEach { (buttonId, productType) ->
            val button = root.findViewById<ImageButton>(buttonId)
            button.setOnClickListener {
                Log.d("HomeFragment", "$productType button clicked")
                val intent = Intent(requireContext(), ProductListActivity::class.java)
                intent.putExtra("productType", productType)
                startActivity(intent)
            }
        }
        return root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }


}